lib/test/integration/sftp_test.go

```diff
@@ -35,6 +49,27 @@ var _ = registerIntegrationTest("sftp", func(t *testing.T) {
 
 	resource.Expire(900)
 
+	creds := sftpSetup.Credentials{
+		Username: sftpUsername,
+		Password: sftpPassword,
+	}
+
+	sftpPort = resource.GetPort("22/tcp")
+	sftpEndpoint := "localhost:" + sftpPort
+	deadline := time.Now().Add(dialTimeout)
+	for time.Now().Before(deadline) {
+		t.Logf("waiting for SFTP server to come up on '%v'...", "localhost:"+sftpPort)
+
+		sftpClient, err = creds.GetClient(sftpEndpoint)
```

Comment:
This should go within the `pool.Retry` block above, it handles looping, backoff, etc for you.